test/test_torch.py

```diff
@@ -16987,6 +16987,23 @@ def test_lcm(self, device, dtype):
         expected = np.lcm(a.cpu().numpy(), b.cpu().numpy())
         self.assertEqual(actual, expected)
 
+    @onlyOnCPUAndCUDA
+    @dtypes(torch.float32, torch.float64)
```

Comment:
`@dtypesIfCUDA(torch.float16, torch.float32, torch.float64)` can be added to test your half implementation on CUDA only